core/src/main/java/org/elasticsearch/search/aggregations/pipeline/bucketscript/BucketScriptPipelineAggregator.java

```diff
@@ -153,22 +161,110 @@ protected void doReadFrom(StreamInput in) throws IOException {
 
     public static class Factory extends PipelineAggregatorFactory {
 
-        private Script script;
-        private final ValueFormatter formatter;
-        private GapPolicy gapPolicy;
-        private Map<String, String> bucketsPathsMap;
+        private final Script script;
+        private final Map<String, String> bucketsPathsMap;
+        private String format = null;
+        private GapPolicy gapPolicy = GapPolicy.SKIP;
```

Comment:
This is a convention which started in the query refactoring so there is a single place for the default values. I also think it actually makes sense for the case where the client are server are not on the same version as the defaults for the client will be retained meaning less surprises when upgrading?